### PR TITLE
Wk 195 unittests for autorization

### PR DESCRIPTION
**Pull request by aditudorache**
This PR adds unittests for the autorization functionality

### CODE REVIEW

File: src/components/AppRoutes.test.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+describe('AppRoutes', () => {
+  beforeEach(() => {
+    DashboardPage.mockReturnValue(<div>dashboard</div>);
+    ConceptPage.mockReturnValue(<div>concept</div>);
+    ContactPage.mockReturnValue(<div>contact</div>);
+    ManageLocationPage.mockReturnValue(<div>manage</div>);
+  });
+
+  afterEach(cleanup);
```
Comment (jasperswart) [inline src/components/AppRoutes.test.jsx]: cleanup is not needed with the latest version

File: src/components/detailPanel/DetailPanel.test.js [hunk 2]
```diff
@@ -1,10 +1,75 @@
...
+    );
+    expect(container.firstChild.innerHTML).toBe('');
+  });
+
+  it('should render the edit link when authorized ', () => {
+    var user = { canEdit: true };
+    const { queryByText } = render(
+      withUserContext(<DetailPanel {...props} />, user)
+    );
+    expect(queryByText(/Wijzig/)).not.toBeNull();
```
Comment (jasperswart) [inline src/components/detailPanel/DetailPanel.test.js]: I think that selector is too wide: this merely says that the text Wijzig should be visible in het whole page.
better to use data-testid="wijzigButton" than you really know if the wijzig button is visible.
the same applies to other tests here.

File: src/components/detailPanel/DetailPanel.test.js [hunk 3]
```diff
@@ -1,10 +1,75 @@
...
+  });
+
+  it('should NOT render the edit link when not authorized for edit ', () => {
+    var user = { canEdit: false };
+    const { queryByText } = render(
+      withUserContext(<DetailPanel {...props} />, user)
+    );
+    expect(queryByText(/Wijzig/)).toBeNull();
```
Comment (jasperswart) [inline src/components/detailPanel/DetailPanel.test.js]: on the detail page I also expect a test to check for all the values like location, status, lat long, opmerkingen etc.

File: src/components/AppRoutes.test.jsx (hunk 1 above)
Comment (aditudorache) [reply src/components/AppRoutes.test.jsx]: thanks!

File: src/components/detailPanel/DetailPanel.test.js (hunk 2 above)
Comment (aditudorache) [reply src/components/detailPanel/DetailPanel.test.js]: done!

File: src/components/detailPanel/DetailPanel.test.js (hunk 3 above)
Comment (aditudorache) [reply src/components/detailPanel/DetailPanel.test.js]: out of scope for the moment, is existing functionality

File: Dockerfile [hunk 4]
```diff
@@ -8,23 +8,21 @@ WORKDIR /app
...
 RUN git config --global url."https://github.com/".insteadOf git@github.com:
 
 COPY package.json package-lock.json /app/
 
-RUN npm config set registry https://repo.datapunt.amsterdam.nl/repository/npm-group/ && \
-    npm --production=false \
-        --unsafe-perm \
-        --verbose \
-        install && \
-    npm cache clean --force
```
Comment (janjaap) [inline Dockerfile]: You should keep this line. It keeps the image size small.

File: Jenkinsfile [hunk 5]
```diff
@@ -22,11 +22,18 @@ node {
...
-    //         sh "api/deploy/test/jenkins-script.sh api/deploy/test/"
-    //     }
-    // }
+    stage('Test') {
+        String PROJECT = "blackspots-unittests-${env.GIT_COMMIT}"
+
+        tryStep "unittests start", {
+            sh "docker-compose -p ${PROJECT} up --build --exit-code-from unittest unittest"
+        }
+        always {
```
Comment (janjaap) [inline Jenkinsfile]: This block will never be called. Do it like this:
```
tryStep "unittests start", {
    sh "docker-compose -p ${PROJECT} up --build --exit-code-from unittest unittest"
}, {
    sh "docker-compose -p ${PROJECT} down -v || true"
}
```

That also goes for the other steps in this file

File: src/test/utils.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { Router } from 'react-router-dom';
+import { createMemoryHistory } from 'history';
+import { UserContextProvider } from 'shared/user/UserContext';
+
+const history = createMemoryHistory();
+
+export const withUserContext = (Component, user, route) => {
```
Comment (janjaap) [inline src/test/utils.js]: No tests for this function?

File: src/views/manage/FormFields.js [hunk 7]
```diff
@@ -55,6 +61,7 @@ export const initalValues = {
 };
 
 export const FormField = ({
+  // key,
```
Comment (janjaap) [inline src/views/manage/FormFields.js]: Forgot to remove

File: src/views/manage/ManageFormComponent.jsx [hunk 8]
```diff
@@ -14,14 +14,15 @@ const ManageFormComponent = ({
 }) => {
   return (
     <ManageFormStyle onSubmit={handleSubmit} action="" novalidate>
-      <Row wrap>
+      <Row>
         <StyledColumn span={6} direction="vertical">
```
Comment (janjaap) [inline src/views/manage/ManageFormComponent.jsx]: `Column` component doesn't have `vertical` prop.

File: Dockerfile (hunk 4 above)
Comment (aditudorache) [reply Dockerfile]: Not sure it is necessary, this image is just the build image, the build results are copyed in the other image, this image is thrown away after each CI build, si not saved in the repository. This cleanup was handy in older versions of the dockerfile when multi images were not possible.

File: Jenkinsfile (hunk 5 above)
Comment (aditudorache) [reply Jenkinsfile]: thanks!

File: src/views/manage/FormFields.js (hunk 7 above)
Comment (aditudorache) [reply src/views/manage/FormFields.js]: done!

File: src/test/utils.js (hunk 6 above)
Comment (aditudorache) [reply src/test/utils.js]: not needed, is just a test helper.

File: src/views/manage/ManageFormComponent.jsx (hunk 8 above)
Comment (aditudorache) [reply src/views/manage/ManageFormComponent.jsx]: removed

**Pull request merged**